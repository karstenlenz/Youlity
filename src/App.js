import React, { useState, useEffect } from 'react'
import Questionnaire from './Questionnaire/Questionnaire'
import { Switch, Route } from 'react-router-dom'
import Motives from './Motives/Motives'
import { evaluateMatchingStyles } from './Motives/util'

function App() {
  const [userMotives, setUserMotives] = useState([])
  const [userStyles, setUserStyles] = useState([])

  useEffect(() => {
    if (userMotives.length === 3) {
      console.log('Matching styles:' + evaluateMatchingStyles(userMotives))
      setUserStyles(evaluateMatchingStyles(userMotives))
    }
  }, [userMotives])

  return (
    <main>
      <Switch>
        <Route path="/questionnaire">
          <Questionnaire userStyles={userStyles} />
        </Route>
        <Route path="/">
          <Motives userMotives={userMotives} setUserMotives={setUserMotives} />
        </Route>
      </Switch>
    </main>
  )
}

export default App
