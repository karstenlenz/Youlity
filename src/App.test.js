import { render } from '@testing-library/react'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import App from './App'

it('renders correctly', () => {
  const { container } = render(
    <Router>
      <App />
    </Router>
  )
  expect(container).toMatchSnapshot()
})
