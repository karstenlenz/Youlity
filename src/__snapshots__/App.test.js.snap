// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <main>
    <h1>
      Bedürfnisse
    </h1>
    <h2>
      Bitte wählen Sie die 3 Bedürfnisse aus, die Ihnen am wichtigsten sind. Fangen Sie mit dem wichtigsten an.
    </h2>
    <ol>
      <li>
         
         
      </li>
      <li>
         
         
      </li>
      <li>
         
         
      </li>
    </ol>
    <section>
      <button>
        Anerkennung
      </button>
      <button>
        Wichtigkeit
      </button>
      <button>
        Verlässlichkeit
      </button>
      <button>
        Solidarität
      </button>
      <button>
        Autonomie
      </button>
      <button>
        Grenzen
      </button>
    </section>
  </main>
</div>
`;
